src/server/script_mgr.cc

```diff
@@ -272,6 +283,16 @@ optional<ScriptMgr::ScriptData> ScriptMgr::Find(std::string_view sha) const {
   return std::nullopt;
 }
 
+void ScriptMgr::FlushAllScript() {
+  lock_guard lk{mu_};
+  db_.clear();
+
+  ServerState* ss = ServerState::tlocal();
+  auto interpreter = ss->BorrowInterpreter();
+  interpreter->ResetStack();
```

Comment:
So as I wrote above the ResetStack does not flushes the scripts data from interpreter.
From what I understand we should actually reset all interpreters in all shards.
To do so we need to do 
  shard_set->pool()->Await([&sha, &params](auto index, auto* pb) {
    ServerState::tlocal()->ResetInterpreters();
  }); 
This will make sure that ResetInterpreters will run on all dragonfly shards
ServerState::ResetInterpreters should access interpreter_mgr_ and reconstruct all the Interpreters in InterpreterManager::storage_. also you will need to make sure that no initerperter is borrowed i.e available_.size() == storage_.size() before you reset them as there might be other lua scripts running when you run flush

Reply:
@adiholden 
Thanks for your help!
I fixed my code according to your advice. 
https://github.com/dragonflydb/dragonfly/pull/2493/commits/ab2662b91288ec3d5c2e3452f769ab36346a6266

Currently, my code occured segmentation fault when running EVALSHA command after running SCRIPT FLUSH command.
I was investigating this problem.